services/authoring/src/search-index/index.ts

```diff
@@ -72,8 +72,8 @@ export default {
     return dependentsPromise.then(({ aliasIndex, dependencyIndex }) => ({ usedBy: dependencyIndex[key] || [], aliases: aliasIndex[key] || [] }));
   },
   refreshIndex: (repoDir) => {
-    indexPromise = refreshIndex(repoDir);
-    manifestPromise = getManifests(repoDir);
+    indexPromise = refreshIndex(repoDir).catch((x=>()=>x)(indexPromise));
```

Comment:
why not simply `.catch(() => indexPromise)`?

Reply:
It'll be a bit ugly because i'll need to define lots of currentBlaPromise.

Reply:
then maybe 
```js
const persistPromise = x => () => x;
indexPromise = refreshIndex(repoDir).catch(persistPromise(indexPromise));
```

Reply:
Maybe:```makeRef(indexPromise)?```